app/views/TemplateView/index.tsx

```diff
@@ -1,25 +1,40 @@
-import React, { useState, useEffect, useCallback } from 'react';
+import React, { useState, useCallback } from 'react';
 import { useSnackbar } from 'notistack';
+import { useQuery, useMutation, useQueryClient } from 'react-query';
 
 import DataTable from '@/components/DataTable';
 import api from '@/services/api';
 import AddEditTemplate from './components/AddEditTemplate';
 import columnDefs from './columnDefs';
 
+const fetchTemplates = async () => api.get('/templates', {}).request();
+
+const deleteTemplate = async (ident: string) => {
+  await api.del(`/templates/${ident}/signature-types`, {}, {}).request();
+  return api.del(`/templates/${ident}`, {}, {}).request();
+};
+
 const TemplateView = (): JSX.Element => {
   const [showDialog, setShowDialog] = useState(false);
-  const [templates, setTemplates] = useState([]);
   const [selectedRow, setSelectedRow] = useState();
-
   const snackbar = useSnackbar();
+  const queryClient = useQueryClient();
 
-  useEffect(() => {
-    const getData = async () => {
-      const templatesResp = await api.get('/templates', {}).request();
-      setTemplates(templatesResp);
-    };
-    getData();
-  }, []);
+  const { data: templates = [] } = useQuery({
+    queryKey: ['templates'],
+    queryFn: fetchTemplates,
+  });
+
+  const deleteMutation = useMutation({
+    mutationFn: deleteTemplate,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['templates'] });
+      snackbar.enqueueSnackbar('Template deleted');
```

Comment:
actually, we should use snackbar.success since we're here, 

`import snackbar from '@/services/SnackbarUtils';`

and then do `snackbar.success('Template deleted');`

same thing with 
`snackbar.error(`Error deleting template: ${err}`);`